MongoDB conceptual document

PrathibaJee opened the pull request.

PrathibaJee committed:
Create MongoDBDesign.iuml

PrathibaJee committed:
Update MongoDBDesign.iuml

PrathibaJee committed:
Update MongoDB.md

PrathibaJee committed:
Update services.yaml

PrathibaJee committed:
Merge branch 'develop' into PrathibaJee/MongoDB

PrathibaJee committed:
Update services.yaml

PrathibaJee committed:
Update profileInstances.yaml

PrathibaJee committed:
Update forwardings.yaml

PrathibaJee committed:
Update forwardings.yaml

PrathibaJee committed:
Update profileInstances.yaml

PrathibaJee committed:
Update ApplicationPattern+services.yaml

PrathibaJee committed:
Update ApplicationPattern+services.yaml

PrathibaJee committed:
Update MongoDB.md

kmohr-soprasteria commented on doc/SharedComponents/MongoDB/MongoDB.md:
```diff
@@ -1,10 +1,112 @@
+MongoDB will thus serve as a **highly performant, update-friendly datastore**, while Elasticsearch can continue to serve its strengths in **search and analytics**.
+
+![your-UML-diagram-name](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/openBackhaul/ApplicationPattern/refs/heads/PrathibaJee/MongoDB/doc/SharedComponents/MongoDB/diagram/MongoDBDesign.iuml)
+
```
maybe you could add a placeholder box in both "Apps with NMDA DS" and "Applications" to indicate that there can be more apps using MongoDB or ElasticSearch and the ones given here are just examples

kmohr-soprasteria commented on doc/SharedComponents/MongoDB/MongoDB.md:
```diff
@@ -1,10 +1,112 @@
-### Purpose
+# Purpose
 
+Our current SDN ApplicationLayer architecture uses Elasticsearch to store and query JSON objects. While Elasticsearch provides powerful capabilities for search and analytics, it is not well-suited for use cases involving frequent updates to JSON documents, especially when those updates involve nested fields(like the one in the new NMDA datastore concepts).
```
there's a blank missing before the (like the one...)

kmohr-soprasteria commented on doc/SharedComponents/MongoDB/diagram/MongoDBDesign.iuml:
```diff
@@ -0,0 +1,67 @@
+@startuml
```
see comment in the md doc

kmohr-soprasteria reviewed (approved):
see comments (nothing serious)

anacabbar commented on spec/ApplicationPattern+services.yaml:
```diff
@@ -91,7 +91,7 @@ clients:
       uuid: xx-1-0-0-es-c-es-1-0-0-000
       index-alias:
-    
+
```
Are 'elasticsearch-client' in line 104 and the 'es' in the uuid in line 105 a typo?

anacabbar commented on doc/SharedComponents/MongoDB/MongoDB.md:
```diff
@@ -1,10 +1,112 @@
+                    "password":   " >>> password <<< " 
+                  }, 
+                "database-name": " >>> Application specific database name <<< ", 
+                "collection-name": ">>> collection name specific to database <<<"
```
Since "collection-name" is not a list, if I were to have several collections in a database, would I then need multiple objects in the "layer-protocol" list for the multiple collections? And would I then need one Ltp entry per Data-base in the CONFIGfile?

PrathibaJee committed:
Update services.yaml to use mongodb-client

Replaced elasticsearch-client with mongodb-client and updated UUID.

PrathibaJee committed:
Update MongoDB.md to clarify MongoDB's role

Clarified MongoDB's role as an update-friendly datastore for NMDA based datastores, emphasizing its efficiency in handling nested JSON updates compared to Elasticsearch.

PrathibaJee committed:
Fix formatting in MongoDB.md

Corrected spacing in the MongoDB documentation.

PrathibaJee committed:
Add collection-list structure to MongoDB documentation

PrathibaJee committed:
Add local-id to mongodb-client collection list
